bakery/keys.go

```diff
@@ -79,25 +78,53 @@ func (k *Key) UnmarshalText(text []byte) error {
 	return nil
 }
 
-// PublicKeyLocator is used to find the public key for a given
-// caveat or macaroon location.
-type PublicKeyLocator interface {
-	// PublicKeyForLocation returns the public key matching the caveat or
-	// macaroon location. It returns ErrNotFound if no match is found.
-	PublicKeyForLocation(loc string) (*PublicKey, error)
+// ThirdPartyInfo holds information on a given third party
+// discharge service.
+type ThirdPartyInfo struct {
+	// PublicKey holds the public key of the third party.
+	PublicKey PublicKey
+
+	// Version holds latest the bakery protocol version supported
+	// by the discharger.
+	Version Version
+}
+
+// ThirdPartyLocator is used to find information on third
+// party discharge services.
+type ThirdPartyLocator interface {
+	// ThirdPartyInfo returns information on the third
+	// party at the given location. It returns ErrNotFound if no match is found.
```

Comment:
I think it's reasonable to document an interface like this, for the user rather than the implementor of the interface. io.Reader (for example, as a canonical example of an interface) documents itself in this way and I think it's better than using "should" and "must" all over the place.